compiler/circle-eval-diff/CMakeLists.txt

```diff
@@ -1,10 +1,25 @@
 set(DRIVER "driver/Driver.cpp")
 
 file(GLOB_RECURSE SOURCES "src/*.cpp")
+file(GLOB_RECURSE TESTS "src/*.test.cpp")
+list(REMOVE_ITEM SOURCES ${TESTS})
+set(TEST_SOURCES
+    "src/Tensor.cpp")
 
 add_executable(circle-eval-diff ${DRIVER} ${SOURCES})
 target_include_directories(circle-eval-diff PRIVATE include)
 
 target_link_libraries(circle-eval-diff arser)
 target_link_libraries(circle-eval-diff safemain)
+target_link_libraries(circle-eval-diff loco)
 target_link_libraries(circle-eval-diff vconone)
+
+if(NOT ENABLE_TEST)
+  return()
+endif(NOT ENABLE_TEST)
+
+nnas_find_package(GTest REQUIRED)
+GTest_AddTest(circle_eval_diff_test ${TESTS} ${TEST_SOURCES})
```

Comment:
Aboud adding `${TEST_SOURCES}`
- `circle-eval-diff` is executable so adding link to this exec looks strange
- so adding the source through `${TEST_SOURCES}` looks unavoidable.

Bu we can think this as optional;
- create `add_library(eval_diff_helper STATIC ${SOURCES})`
- add `target_link_libraries(circle-eval-diff eval_diff_helper)`
- and add `target_link_libraries(circle_eval_diff_test eval_diff_helper)` too

Reply:
I've thought about that option, but I decided not to choose it.

`{SOURCES}` have all cpp files, which may require arser, safemain, vconone, which are not used by `circle_eval_diff_test`. To remove those dependencies, we'll have to select source files, similarly to the current way.

Reply:
Ah, OK then can you add some comments of this background facts?